pkg/repositories/gormimpl/artifact.go

```diff
@@ -65,18 +65,20 @@ func (h *artifactRepo) Get(ctx context.Context, in models.ArtifactKey) (models.A
 		)
 
 	if result.Error != nil {
-		return models.Artifact{}, h.errorTransformer.ToDataCatalogError(result.Error)
-	}
-	if result.RecordNotFound() {
-		return models.Artifact{}, errors.GetMissingEntityError("Artifact", &datacatalog.Artifact{
-			Dataset: &datacatalog.DatasetID{
-				Project: in.DatasetProject,
-				Domain:  in.DatasetDomain,
-				Name:    in.DatasetName,
-				Version: in.DatasetVersion,
-			},
-			Id: in.ArtifactID,
-		})
+		switch result.Error.Error() {
```

Comment:
do you need a switch here? why not just an if statement?

Reply:
Both should work. If we want to handle a different error type, probably less code change needed.

Reply:
hm switch seems like overkill still and this code doesn't really change

Reply:
Fixed.